File: lesson4/python3/homework/homework4_21.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_21.py
# a         | 64168
# b         | 82678
# НОД(a, b) | 1234
# шаги-1    | 12
# шаги-2    | 4

numb1 = 64168
numb2 = 82678

count1 = 0
while numb1 != numb2:
    if numb1 > numb2:
        numb1 -= numb2
    else:
        numb2 -= numb1
    count1 += 1
print("numb1 =>", numb1)
print("numb2 =>", numb2)
print("count1 =>", count1)
print()

numb3 = 64168
numb4 = 82678

count2 = 0
while numb3 != 0 and numb4 != 0:
    if numb3 > numb4:
        numb3 %= numb4
    else:
        numb4 %= numb3
    count2 += 1
print("numb3 =>", numb3)
print("numb4 =>", numb4)
print("count2 =>", count2)

File: lesson4/c_c++/homework/homework4_21.cpp
//: lesson4/c_c++/homework/homework4_21.cpp
// a         | 64168
// b         | 82678
// НОД(a, b) | 1234
// шаги-1    | 12
// шаги-1    | 4
#include <iostream>

int main() {
    int numb1 = 64168;
    int numb2 = 82678;

    int count1 = 0;
    while (numb1 != numb2) {
        if (numb1 > numb2) {
            numb1 = numb1 - numb2;
        } else {
            numb2 = numb2 - numb1;
        }
        ++count1;
    }
    std::cout << "numb1_end => " << numb1 << std::endl;
    std::cout << "numb2_end => " << numb2 << std::endl;
    std::cout << "count1 => " << count1 << std::endl;

    int numb3 = 64168;
    int numb4 = 82678;

    int count2 = 0;
    while (numb3 != 0 && numb4 != 0) {
        if (numb3 > numb4) {
            numb3 %= numb4;
        } else {
            numb4 %= numb3;
        }
        ++count2;
    }
    std::cout << "numb3_end => " << numb3 << std::endl;
    std::cout << "numb4_end => " << numb4 << std::endl;
    std::cout << "count2 => " << count2 << std::endl;
}
